File: core/src/main/java/ru/surfstudio/android/core/ui/base/screen/model/state/LoadState.java
package ru.surfstudio.android.core.ui.base.screen.model.state;


public enum LoadState {
    UNSPECIFIED, //не определен
    NONE, //контент загружен
    MAIN_LOADING, //прогресс, закрывающий весь контент
    TRANSPARENT_LOADING, //полупрозрачный прогресс, блокирует весь интерфейс
    SMALL_LOADING, //индикатор загрузки, не блокирующий интерфейс, может быть ptr
    ERROR, //ошибка загрузки данных
    EMPTY, //данных нет
    NOT_FOUND, //нет данных по заданному фильтру
}

File: template/base_feature/src/main/java/ru/surfstudio/standard/ui/placeholder/LoadState.kt
package ru.surfstudio.standard.ui.placeholder

import ru.surfstudio.android.core.mvp.loadstate.LoadStateInterface

class NoneLoadState : LoadStateInterface
class ErrorLoadState : LoadStateInterface
class EmptyLoadState : LoadStateInterface
class NotFoundLoadState : LoadStateInterface
class MainLoadingState : LoadStateInterface
class TransparentLoadingState : LoadStateInterface
class NoInternetLoadState : LoadStateInterface

object LoadState {
    val NONE = NoneLoadState()
    val MAIN_LOADING = MainLoadingState()
    val TRANSPARENT_LOADING = TransparentLoadingState()
    val ERROR = ErrorLoadState()
    val EMPTY = EmptyLoadState()
    val NOT_FOUND = NotFoundLoadState()
    val NO_INTERNET = NoInternetLoadState()
}